unity-renderer/Assets/UIComponents/Scripts/Components/CollapsableSortedList/CollapsableSortedListComponentView.cs

```diff
@@ -0,0 +1,139 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using DCL.Helpers;
+using UnityEngine;
+
+namespace UIComponents.CollapsableSortedList
+{
+    public class CollapsableSortedListComponentView<K, V> : BaseComponentView
+        where V : Component
+    {
+        private readonly Dictionary<K, V> entries = new Dictionary<K, V>();
+        private readonly List<K> sortedEntries = new List<K>();
+
+        [SerializeField] private Transform container;
+        [SerializeField] private CollapsableListToggleButton toggleButton;
+        [SerializeField] private GameObject emptyStateContainer;
+        [SerializeField] private CollapsableSortedListModel model;
+
+        private int filteredCount;
+
+        public Dictionary<K, V> Entries => entries;
+
+        public Comparison<V> SortingMethod { get; set; } = (model, model1) => 0;
+
+        public override void OnEnable()
+        {
+            base.OnEnable();
+            UpdateEmptyState();
+        }
+
+        public override void RefreshControl()
+        {
+            Clear();
+            
+            if (model.isExpanded)
+                Expand();
+            else
+                Collapse();
+
+            if (model.isVisible)
+                Show();
+            else
+                Hide();
+        }
+
+        public virtual bool Contains(K key) => entries.ContainsKey(key);
+
+        public virtual void Expand()
+        {
+            toggleButton.Toggle(true);
+            model.isExpanded = true;
+        }
+
+        public virtual void Collapse()
+        {
+            toggleButton.Toggle(false);
+            model.isExpanded = false;
+        }
+
+        public virtual int Count() => entries.Count - filteredCount;
+
+        public virtual void Show()
+        {
+            gameObject.SetActive(true);
+            model.isVisible = true;
+        }
+
+        public virtual void Hide()
+        {
+            gameObject.SetActive(false);
+            model.isVisible = false;
+        }
```

Comment:
These `Show()` / `Hide()` functions... shouldn't be overriden from the `BaseComponentView` class?

Reply:
done!